crates/gpui/src/platform/windows/util.rs

```diff
@@ -1,5 +1,132 @@
+use std::cell::Cell;
+
 use windows::Win32::{Foundation::*, UI::WindowsAndMessaging::*};
 
+use crate::{GlobalPixels, Pixels, Point, Size};
+
+type PhysicalPoint = Point<GlobalPixels>;
+pub(crate) type LogicalPoint = Point<Pixels>;
+type PhysicalSize = Size<GlobalPixels>;
+type LogicalSize = Size<Pixels>;
+
+#[derive(Debug, Default)]
+pub(crate) struct WindowMetrics {
+    origin: Cell<PhysicalPoint>,
+    physical_size: Cell<PhysicalSize>,
+    logical_size: Cell<LogicalSize>,
+    scale_factor: Cell<f32>,
+}
+
+impl WindowMetrics {
```

Comment:
I think that, rather than creating this type, we should just be consistently using the existing GPUI types, Pixels, GlobalPixels, ScaledPixels, DevicePixels, Size<T>, etc.

Reply:
Using these aliases might make things clearer, I guess. It was only after reading the macOS impls that I understood the things realy goes here. Should we remove the type aliases and keep this struct? Any suggestions?

Reply:
Unless there's a compelling code reason to do otherwise, I think the best solution would be to take all of these type aliases and the WindowMetrics struct out, and to just have `origin: Point<GlobalPixels>`, `size: Size<GlobalPixels>`, and `scale_factor: f32`. As that's the only data you actually need to store, the logical size can be recomputed on the fly. 

Once you have those inlined, I think you could just make a `logical(x, y, scale_factor) -> Point<Pixels>` helper that you use when you need it :)

Reply:
Makes sense.